orchestrator-salt/src/main/resources/salt/salt/postgresql/disaster_recovery/scripts/backup_db_compressed.sh

```diff
@@ -0,0 +1,254 @@
+  elif kinit_as solr "$SOLR_KEYTAB"; then
+    doLog "Successful kinit using solr principal"
+  elif kinit_as atlas "$ATLAS_KEYTAB"; then
+    doLog "Successful kinit using atlas principal"
+  elif kinit_as knox "$KNOX_KEYTAB"; then
+    doLog "Successful kinit using knox principal"
+  fi
+}
+
+replace_ranger_group_before_export() {
+  doLog "Ranger admin group is $RANGERGROUP"
+  doLog "INFO Replacing "$1" with _RANGER_WAG_2f0264fa-0a04-462b-af85-7c09891568ef in the dump before export"
+  ret_code=$(sed --in-place="_original" -e s/"$1"/_RANGER_WAG_2f0264fa-0a04-462b-af85-7c09891568ef/g "$2" || echo $?)
+  if [[ -n "$ret_code" ]] && [[ "$ret_code" -ne 0 ]]; then
+    errorExit "Unable to re-write file $2"
+  else
+    rm "${2}_original"
+  fi
+}
+
+move_backup_to_cloud () {
+  LOCAL_BACKUP="$1"
+  REMOTE_BACKUP_NAME="$2"
+  run_kinit
+  doLog "INFO Uploading to ${BACKUP_LOCATION}"
+
+  hdfs --loglevel ERROR dfs -mkdir -p "$BACKUP_LOCATION"
+  OBJECT_STORE_PATH="${BACKUP_LOCATION}/${REMOTE_BACKUP_NAME}"
+  hdfs --loglevel ERROR dfs -moveFromLocal -f "$LOCAL_BACKUP" "$OBJECT_STORE_PATH" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to upload $SERVICE backup"
+  doLog "INFO Completed upload to ${BACKUP_LOCATION}"
+}
+
+close_existing_connections() {
+  SERVICE=$1
+  doLog "INFO Closing existing connections to ${SERVICE}"
+  psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "SELECT pg_terminate_backend(pg_stat_activity.pid) FROM pg_stat_activity WHERE pg_stat_activity.datname = '${SERVICE}' AND pid <> pg_backend_pid();" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to close connections to ${SERVICE}"
+}
+
+limit_incomming_connection() {
+  SERVICE=$1
+  COUNT=$2
+  doLog "INFO limit existing connections to ${COUNT} on ${SERVICE}"
+  psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "alter user ${SERVICE} connection limit ${COUNT};" > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to limit connections to ${SERVICE}"
+}
+
+is_database_exists() {
+  SERVICE=$1
+  doLog "INFO Checking the existence of database ${SERVICE}"
+  database_name=$(psql --host="$HOST" --port="$PORT" --dbname="postgres" --username="$USERNAME" -c "SELECT datname FROM pg_catalog.pg_database WHERE datname = '$SERVICE';" -At > >(tee -a $LOGFILE) 2> >(tee -a $LOGFILE >&2) || errorExit "Unable to check the existence of database $SERVICE")
+  if [[ "$database_name" != "$SERVICE" ]];then
+    return 1
+  fi
+  return 0
+}
+
+backup_database_for_service() {
+  SERVICE="$1"
+  is_database_exists $SERVICE
+  if [ "$?" -eq 1 ];then
+    doLog "WARN database for $SERVICE doesn't exist"
+    return 0
+  fi
+
+  if [[ "$CLOSECONNECTIONS" == "true" ]]; then
+    limit_incomming_connection $SERVICE 0
+    close_existing_connections $SERVICE
+  fi
+
+  doLog "INFO Dumping ${SERVICE}"
+  LOCAL_BACKUP=${DATE_DIR}/${SERVICE}_backup
+  if [[ "$SERVICE" == "ranger" ]]; then
```

Comment:
:point_up: never compress Ranger since we do text modifications on the backup script afterwards.